### ISSUE TITLE
PPC lbzu    r9, 1(r4) instruction does not update r4

### ISSUE THREAD

**Issue by TommyBurger** [radareorg/cutter#1832]
**Environment information**
* Operating System: Ubuntu Linux 16.04
* Cutter version:
Version 1.9.0 (built from source on 10/9/2019)
Using r2-3.9.0-137-gf6036ff
* File format: elf

**Describe the bug**
Using the following c source:
//powerpc-linux-gnu-gcc test.c -Os -nostdlib
void func(char *p1, char *p2)
{
  while(*p1++ = *p2++)
    ;
}
int _start()
{
  char buff[64];
  func(buff, "Hello, World.");
}

willl generate object code that includes:

28: sym.func ();
0x100000b8      addi    r4, r4, 0xffff ; [02] -r-x section size 72 named .text
0x100000bc      addi    r3, r3, 0xffff
0x100000c0      lbzu    r9, 1(r4)
0x100000c4      stbu    r9, 1(r3)
0x100000c8      cmpwi   cr7, r9, 0
0x100000cc      bne     cr7, 0x100000c0
0x100000d0      blr

The lbzu and stbu instructions should update r3 and r4 with the new effective address. It does not and ends up looping forever.

**To Reproduce**

1. Create a file from the above source.
2. Compile with powerpc-linux-gnu-gcc test.c -Os -nostdlib
3. Open the resulting a.out with Cutter
4. Start emulation in Cutter.
5. Step until lbzu and stbu instructions.
6. Observe r3 and r4 do not change and loops forever.

**Comment by ITAYC0HEN** [comment]
Thank you for reporting this issue! this problem is in radare2, and reflects in Cutter. I'll transfer the issue to the radare2 repository if it's okay to you 

may I ask you to attach the binary? for those of you who doesn't have a suitable environment

Thanks!

**Comment by TommyBurger** [comment]
Yes. I was not sure where the problem was, so I started here with Cutter. Here is a.out. Note that I corrected the bug report. -Og does not generate the lbzu instruction. I compiled with -Os.
[a.out.zip](https://github.com/radareorg/cutter/files/3712704/a.out.zip)
